### ISSUE TITLE
Fix S3878 FP: When input array is collection expression with spread operator

### ISSUE THREAD

**Issue by mary-georgiou-sonarsource** [SonarSource/sonar-dotnet#8510]
### Description

The rule should not raise an issue for collection expression with the spread operator. The `[a, .. bs]` expression below is equivalent to `array.Prepend(a).ToArray()` and cannot be replaced with `a,...,array` as it's a compile error.

### Repro steps

```csharp
using System;

const int a = 1;
int[] array = [2, 3];
_ = new MyClass([a, .. array]);  // Noncompliant FP

void  Method(params int[] args) { } 
```

**Comment by Guiorgy** [comment]
`S3878` is also reported when using a different collection type on the caller side:

```cs
public void MethodWithParams(params string[] args)
{
    // do something
}

public void MethodPassingList()
{
    var list = new List<string>();
    MethodWithParams([.. list]); // Warning	S3878	Remove this array creation and simply pass the elements.
}
```

Passing `list.ToArray()` makes Visual Studio advise the use of the collection expression (`Use collection expression for fluent (IDE0305)`), which refactors to the above.

**Comment by mary-georgiou-sonarsource** [comment]
Another report for this issue [here](https://discuss.sonarsource.com/t/false-positive-with-s3878-in-c/17961) .

**Comment by Guiorgy** [comment]
@mary-georgiou-sonarsource BTW, should my case (comment above) be a separate issue?

**Comment by mary-georgiou-sonarsource** [comment]
@Guiorgy  In my opinion no, because the main idea is that we don't raise when the spread operator is used (like we do for `ToArray()`).
